### PR TITLE
Getting Urltofile to return a wtype.file

### PR DESCRIPTION
**Pull request by cachemoi**
previous pull request had some troubles with types

### CODE REVIEW

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 1]
```diff
@@ -2,30 +2,41 @@
...
-	if _, err := io.Copy(f, res.Body); err != nil {
-		return err
+	//converting body to bytes
+	if resp.StatusCode == 200 { // OK
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return file, err
+		}
+		buf.Write(bodyBytes)
 	}
```
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: else if resp.StatusCode != 200 {
// return an error? retry?
}

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 2]
```diff
@@ -2,30 +2,41 @@
...
 import (
-	"io"
 	"net/http"
-	"os"
+	"github.com/antha-lang/antha/antha/anthalib/wtype"
+	"bytes"
+	"io/ioutil"
 )
 
 // File downloads the data at a url to the given filename. If there is an error, the file will contain the partially downloaded data.
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: The second part of this comment is not correct according to the implementation below

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 3]
```diff
@@ -2,30 +2,41 @@
...
-		return err
+		return file, err
 	}
-	defer f.Close()
+	defer resp.Body.Close()
 
-	if _, err := io.Copy(f, res.Body); err != nil {
-		return err
+	//converting body to bytes
+	if resp.StatusCode == 200 { // OK
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: if resp.StatusCode == http.StatusOK

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 4]
```diff
@@ -2,30 +2,41 @@
...
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: _, err := io.Copy(&buf, resp.Body)

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 5]
```diff
@@ -2,30 +2,41 @@
...
+		if err != nil {
+			return file, err
+		}
+		buf.Write(bodyBytes)
 	}
 
-	return nil
-}
+	//returning wtype.File
+	file.WriteAll(buf.Bytes())
```
Comment (ddn0) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: if err := file.WriteAll(buf.Bytes()); err != nil { ... }

**Comment by chr1sgrant** [comment]
@synthacebot  test this please

**Comment by chr1sgrant** [comment]
Looks like I still can't trigger CI testing on PRs submitted by collaborators

**Comment by ddn0** [comment]
@synthacebot test this please

**Comment by cachemoi** [comment]
@ddn0 Added the changes you requested. Rewrote comments to be accurate

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 6]
```diff
@@ -1,31 +1,48 @@
...
 	"net/http"
-	"os"
+	"github.com/antha-lang/antha/antha/anthalib/wtype"
+	"bytes"
+	"io"
 )
 
-// File downloads the data at a url to the given filename. If there is an error, the file will contain the partially downloaded data.
-func File(url string, filename string) (err error) {
+func File(url string, fileName string) (file wtype.File, err error) {
```
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: Should be commented above function name to autogenerate documentaiton via godocs

//Given a URL and a desired file name, this function will return the whole content of the response into a wtype.DownloadFile object.
func File (...

**Comment by m-i-sadowski** [comment]
@synthacebot test this please

File: antha/AnthaStandardLibrary/Packages/download/urltofile.go [hunk 7]
```diff
@@ -10,13 +10,10 @@ import (
 	"io"
 )
 
+//File takes a URL and a desired file name, and returns the whole content of the response into a wtype.DownloadFile object.
```
Comment (chr1sgrant) [inline antha/AnthaStandardLibrary/Packages/download/urltofile.go]: //File takes a URL and a desired file name, and returns the whole content of the response into a wtype.File object.

**Comment by chr1sgrant** [comment]
@synthacebot test this please

**Comment by chr1sgrant** [comment]
@synthacebot test this please

**Comment by cachemoi** [comment]
Moved to origin/urltofileRefractor

**Pull request closed without merging**